### PR TITLE
[DNM][RFC] Push atmos adjacency calculation to a new subsystem

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
This improves init times and makes explosions and other shit that modifies a bunch of turfs faster by saving a proccall and potentially not doing the same work more than once. 

:cl: Naksu
code: Turf adjacency list updates are now handled by a subsystem queue
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/adjacent_air.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
+SUBSYSTEM_DEF(adjacent_air)
+	name = "Atmos Adjacency"
+	flags = SS_BACKGROUND
+	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+	wait = 1
```
Comment (vuonojenmustaturska) [inline code/controllers/subsystem/adjacent_air.dm]: Gonna need to test this a bit more and find a good number here, 5-10 is probably appropriate. The higher it is the more of the potential gains during explosions and such can be realized, but making it too high turns rooms into ZAS zones

Comment (Poojawa) [reply code/controllers/subsystem/adjacent_air.dm]: Some ZAS performance is okay, after all high pressure gas propagates extremely quickly into rooms. aka 4.500 kPa etc.

Comment (duncathan) [reply code/controllers/subsystem/adjacent_air.dm]: Experiment with as high as 10-15 imo I'm interested in some very minor zone behavior

**Review by duncathan — DISMISSED** [review]
I'm excited to see how this plays out

File: code/controllers/subsystem/adjacent_air.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				if (neighbor_adjacent_turfs)
+					neighbor_adjacent_turfs -= currT
+				UNSETEMPTY(neighbor_adjacent_turfs)
+			neighborT.atmos_adjacent_turfs = neighbor_adjacent_turfs
+		UNSETEMPTY(atmos_adjacent_turfs)
+		// If the list was null before, our thing isn't a reference to the thing in the object, we need to assign the list back just in case.
+		currT.atmos_adjacent_turfs = atmos_adjacent_turfs
+
+		if (MC_TICK_CHECK)
+			return
```
Comment (Cyberboss) [inline code/controllers/subsystem/adjacent_air.dm]: This needs a regular CHECK_TICK as well if it's running in Initialize. See SSoverlays/fire

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Qustinnus** [comment]
whats the actual diff in init time

**Comment by vuonojenmustaturska** [comment]
i haven't forgotten about this, i'm just juggling potential approaches to some of the issues this has exposed

**Comment by vuonojenmustaturska** [comment]
keeping /turf/air_update_turf immediate still saves some calculations (5k opposed to 15k roundstart) and some for explosions/shuttles without having obvious issues with shuttles, there's probably an optimization waiting to be done to get some of the savings back.

**Comment by vuonojenmustaturska** [comment]
causes runtimes on testmerge, maybe it was not meant to be

**Pull request closed without merging**